sno/working_copy/sqlserver_adapter.py

```diff
@@ -0,0 +1,179 @@
+from sno import crs_util
+from sno.schema import Schema, ColumnSchema
+
+from sqlalchemy.sql.compiler import IdentifierPreparer
+from sqlalchemy.dialects.mssql.base import MSDialect
+
+
+_PREPARER = IdentifierPreparer(MSDialect())
+
+
+def quote(ident):
+    return _PREPARER.quote(ident)
+
+
+V2_TYPE_TO_MS_TYPE = {
+    "boolean": "bit",
+    "blob": "varbinary",
+    "date": "date",
+    "float": {0: "real", 32: "real", 64: "float"},
+    "geometry": "geography",
```

Comment:
```suggestion
    "geometry": "geometry",
```

Reply:
Not done - not sure if we should be using the SQL server geometry type, geography type, or if it should be configurable. (Same with VARCHAR vs NVARCHAR). Have you looked into this?

I tried switching geography for geometry right now and the tests failed since SQL server can't create a spatial index for a geometry type without me specifying the bounding box of the spatial index, apparently - I guess for a Geography type, Earth itself is used as the bounding box.

Reply:
Not done. 

I'm not sure which SQL server type is appropriate - geometry or geography, or if it should be user-configurable. (Similarly: VARCHAR vs NVARCHAR).

I tried switching it to geometry, but it caused tests to fail since it could no longer create a spatial index - it says something about how creating spatial index for the geometry type column would require me to specify which bounding box the index should be created over. (I assume this is not required for geograpy types, since it just uses the whole world? Not sure.)

Anyway, I left it as is for now. Have you looked into the geometry and geography types?

Reply:
Switched to geometry and added a heuristic for selecting the spatial index bounding box, as discussed offline.

Reply:
> (Similarly: VARCHAR vs NVARCHAR).

From a bit of reading, seems like `NVARCHAR` is normal for unicode pre- MSSQL 2019, and [from then `VARCHAR` explicitly supports UTF8](https://docs.microsoft.com/en-us/sql/t-sql/data-types/char-and-varchar-transact-sql?view=sql-server-ver15). So I think using `NVARCHAR`/`NCHAR`/`NTEXT` makes sense initially, and we can revisit in the future.

Reply:
Marking this as resolved